src/solace_ai_connector/components/general/llm/litellm/litellm_base.py

```diff
@@ -0,0 +1,349 @@
+            },
+            "first_chunk": {
+                "type": "boolean",
+                "description": "Whether this is the first chunk of the response",
+            },
+            "last_chunk": {
+                "type": "boolean",
+                "description": "Whether this is the last chunk of the response",
+            },
+            "streaming": {
+                "type": "boolean",
+                "description": "Whether this is a streaming response",
+            },
+        },
+        "required": ["content"],
+    },
+}
+
+
+class LiteLLMChatModelBase(ComponentBase):
+    def __init__(self, module_info, **kwargs):
+        super().__init__(module_info, **kwargs)
+        self.init()
+        self.init_load_balancer()
+
+    def init(self):
+        litellm.suppress_debug_info = True
+        self.action = self.get_config("action")
+        self.load_balancer = self.get_config("load_balancer")
+        self.stream_to_flow = self.get_config("stream_to_flow")
+        self.stream_to_next_component = self.get_config("stream_to_next_component")
+        self.llm_mode = self.get_config("llm_mode")
+        self.stream_batch_size = self.get_config("stream_batch_size")
+        self.set_response_uuid_in_user_properties = self.get_config(
+            "set_response_uuid_in_user_properties"
+        )
+        if self.stream_to_flow and self.stream_to_next_component:
+            raise ValueError(
+                "stream_to_flow and stream_to_next_component are mutually exclusive"
+            )
+        self.router = None
+
+    def init_load_balancer(self):
+        """initialize a load balancer"""
+        try:
+            self.router = litellm.Router(model_list=self.load_balancer)
+            log.debug("Load balancer initialized with models: %s", self.load_balancer)
+        except Exception as e:
+            raise ValueError(f"Error initializing load balancer: {e}")
+    
+    def load_balance(self, messages, stream):
+        """load balance the messages"""
+        response = self.router.completion(model=self.load_balancer[0]["model_name"], 
+                messages=messages, stream=stream)
+        log.debug("Load balancer response: %s", response)
+        return response
+
+    def invoke(self, message, data):
+        """invoke the model"""
+        messages = data.get("messages", [])
+
+        if self.action == "inference":
+            if self.llm_mode == "stream":
+                return self.invoke_stream(message, messages)
+            else:
+                return self.invoke_non_stream(messages)
+        elif self.action == "embedding":
+            return self.invoke_embedding(messages)
+        else:
+            raise ValueError(f"Unsupported action: {self.action}") 
+
+    def invoke_embedding(self, messages):
+        """invoke the embedding model"""
+        response = self.router.embedding(model=self.load_balancer[0]["model_name"], 
+                input=messages[0]["content"])
```

Comment:
Embedding shouldn't be using the same interface as LLM invoke.

Embedding requires an array of items